test/parallel/test-fs-mkdir.js

```diff
@@ -132,6 +132,24 @@ function nextdir() {
   }
 }
 
+// mkdirpSync when part of the path is a file.
+{
+  const filename = path.join(tmpdir.path, nextdir(), nextdir());
+  const pathname = path.join(filename, nextdir(), nextdir());
+
+  fs.mkdirSync(path.dirname(filename));
+  fs.writeFileSync(filename, '', 'utf8');
+
+  try {
+    fs.mkdirSync(pathname, { recursive: true });
+    throw new Error('unreachable');
+  } catch (err) {
+    assert.notStrictEqual(err.message, 'unreachable');
+    assert.strictEqual(err.code, 'ENOTDIR');
+    assert.strictEqual(err.syscall, 'mkdir');
+  }
```

Comment:
I suggest to use `assert.throws` in this case (hint: when using the error object, it's possible to use regular expressions to verify for the properties. That would probably be handy for `message`).

Reply:
I based the tests on the existing tests (like the one immediately above this one). Perhaps it's better to keep them consistent and then have another PR that changes all of them (including the tests not touched by this PR) to use `assert.throws`?

Reply:
I don't think it's really that urgent that we have to have a follow up PR, even if not changed. If you want to have it consistent and want to change it, I would just do it in this PR.

Reply:
I'm going to leave them as-is for now. `test-fs-promises.js` have similar tests that use `await` which I think means they cannout use `assert.throws`?

Reply:
For promises we have to use `assert.rejects` instead.

Reply:
Pushed a commit to use `assert.rejects()` and `assert.throws()`. PTAL.